### PR TITLE
Fix django 1.8 RequestSite error on LoginView

### PR DESCRIPTION
**Pull request by tony**
Django 1.8 will break if using sites framework

- https://docs.djangoproject.com/en/1.8/ref/contrib/sites/#get-current-site-shortcut
- https://github.com/django/django/blob/1.8/django/contrib/sites/models.py#L50

When using django 1.8 on settings without a ``SITE_ID``, an exception will be raised.

Tthis is permitted in 1.8, since django introduces [get_current_site](https://docs.djangoproject.com/en/1.8/ref/contrib/sites/#get-current-site-shortcut) support in ``get_current`` method in the Manager through the newly-added ``request`` kwarg.

   > Changed in Django 1.8:
   > This function will now lookup the current site based on request.get_host() if the SITE_ID setting is not defined.

Backward compatible with < 1.8.

### CODE REVIEW

**Comment by pennersr** [comment]
There are more places where the site is being looked up. Could we move your lookup into a new to be introduced `def get_current_site(request=None)` method in `allauth.utils`, and then change all lookups to use that one method?

**Comment by tony** [comment]
@pennersr I will follow up tomorrow.

Best

**Comment by tony** [comment]
@pennersr : How is it now?  Would you like it renamed to ``get_current_site``? I used ``get_current_site_model`` to get consistent with ``get_user_model``.

**Comment by pennersr** [comment]
Thanks! If you could rename to `get_current_site` than I can merge it in. The method returns the site instance, whereas `get_user_model` actually returns the model. Hence the naming difference.

**Comment by tony** [comment]
@pennersr Done

File: allauth/socialaccount/views.py [hunk 1]
```diff
@@ -53,7 +52,7 @@ def form_valid(self, form):
 
     def get_context_data(self, **kwargs):
         ret = super(SignupView, self).get_context_data(**kwargs)
-        ret.update(dict(site=Site.objects.get_current(),
+        ret.update(dict(site=get_current_site(),
```
Comment (pennersr) [inline allauth/socialaccount/views.py]: Here you can pass `self.request` as well...

File: example/example/demo/models.py [hunk 2]
```diff
@@ -13,9 +13,11 @@ def setup_dummy_social_apps(sender, **kwargs):
     `allauth` needs tokens for OAuth based providers. So let's
     setup some dummy tokens
     """
-    site = Site.objects.get_current()
+    request = None if 'request' not in kwargs else kwargs['request']
```
Comment (pennersr) [inline example/example/demo/models.py]: How about simply: `request = kwargs.get('request')`

**Comment by tony** [comment]
@pennersr now?

**Comment by pennersr** [comment]
Great, thanks for tweaking ...

**Pull request merged**